trestle/core/catalog_interface.py

```diff
@@ -178,30 +182,44 @@ def get_sorted_controls_in_group(self, group_id: str) -> List[cat.Control]:
         return sorted(controls, key=lambda control: ControlIOWriter.get_sort_id(control))
 
     def get_dependent_control_ids(self, control_id: str) -> List[str]:
-        """Find all children of this control."""
+        """Find all child ids of this control from the dict with recursion."""
         children: List[str] = []
         control = self.get_control(control_id)
-        if control.controls:
-            new_controls = self._get_all_controls_in_list(control.controls, True)
+        if control:
+            new_controls = self._get_all_controls_in_list(as_list(control.controls), True)
             children.extend([con.id for con in new_controls])
         return children
 
     def get_control_ids(self) -> List[str]:
-        """Get all control ids in catalog using the dict."""
+        """Get all control ids from the control dict."""
         return self._control_dict.keys()
 
     def get_control(self, control_id: str) -> Optional[cat.Control]:
-        """Get control from catalog with this id using the dict."""
+        """Get control from the control dict with this id."""
         return None if control_id not in self._control_dict else self._control_dict[control_id].control
 
+    @staticmethod
+    def get_control_ids_from_catalog(catalog: cat.Catalog) -> List[str]:
```

Comment:
I think adding a static function that takes catalog is a bit of overlap with the idea behind `CatalogInterface`. 
i.e.
What is the difference between this one and simply doing this:
```
cat_inter = CatalogInterface(catalog)
cat_inter.get_control_ids()
```

Reply:
This is part of the re-working that was done.  Some interface queries rely on information in the control dictionary - which comprises controls pulled out of the catalog into a separate standalone dictionary.  So for performance, queries against that dict are fast.  So an interface has added expense of needing to crate that dict.  But if I just want to know the controls in a given catalog, a one-pass recursive approach is best.  So it's a matter of doing a one-off scan of controls, vs. a series of queries that justifies the expense of building the dict.

Reply:
So would it be true to say that `cat_inter.get_control_ids() ==  CatalogInterface. get_control_ids_from_catalog(catalog)` ?

Reply:
The results are the same, but there is big expense in creating the control dict just for the purpose of getting the list of ids.  In my usage I need the control dict for one purpose - and I need to know the control ids in the actual catalog.  I have no need to instantiate a new catalog interface on top of the one I already have.  In addition I expect there will be other needs for a quick list of ids in a catalog - without other queries that warrant a cataloginterface.

Reply:
okay, I just find the overhead of maintaining two separate functions with similar functionalities to be not ideal. But given that there is a purpose should be okay, let it stay this way